src/com/dotcms/rest/BaseRestPortlet.java

```diff
@@ -184,24 +187,74 @@ public Response getLayout ( @Context HttpServletRequest request, @Context HttpSe
 
 	}
 
-	private class ResponseWrapper extends HttpServletResponseWrapper {
-
-		private StringWriter writer = new StringWriter();
+	public class ResponseWrapper extends HttpServletResponseWrapper{
```

Comment:
`private static class ...` ditto for `FilterServletOutputStream`
I'm super concerned about this class ever seeing heavy use, as it buffers the entire response. Maybe add a comment pointing out that it's totally going to kill memory use if used to serve large pages?

Also: 
  + What happens if half the content is written to the `writer` and half to `output`?
  + Should we lazy or pre-init the `FilterServletOutputStream` and the `PrintWriter` instead of new-ing one up on each call to the respective getter?